### ISSUE TITLE
Issues with Netmiko 4.x

### ISSUE THREAD

**Issue by nniehoff** [nautobot/nautobot-plugin-device-onboarding#44]
### Environment
* Python version:  3.7
* Nautobot version:  1.3.9
* nautobot-device-onboarding version:  1.1.1

Even though `napalm = ">=2.5.0, <4"` is listed as a dependency somehow I managed to get netmiko 4 installed.  This causes issues with device onboarding.

### Steps to Reproduce
1.  Install Nautobot 1.3.9
2.  Install netmiko 4. (I'm not really sure what other plugin I have that brought this dependency in)
3.  Install device onboarding plugin

### Expected Behavior

should work 

### Observed Behavior

Nautobot fails to start:

```
  File "/usr/local/bin/nautobot-server", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/nautobot/core/cli.py", line 61, in main
    initializer=_configure_settings,  # Called after defaults
  File "/usr/local/lib/python3.7/site-packages/nautobot/core/runner/runner.py", line 266, in run_app
    management.execute_from_command_line([runner_name, command] + command_args)
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.7/site-packages/django/core/management/__init__.py", line 395, in execute
    django.setup()
  File "/usr/local/lib/python3.7/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python3.7/site-packages/django/apps/registry.py", line 122, in populate
    app_config.ready()
  File "/usr/local/lib/python3.7/site-packages/nautobot/extras/plugins/__init__.py", line 104, in ready
    urlpatterns = import_object(f"{self.__module__}.urls.urlpatterns")
  File "/usr/local/lib/python3.7/site-packages/nautobot/extras/plugins/utils.py", line 45, in import_object
    spec.loader.exec_module(module)
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/usr/local/lib/python3.7/site-packages/nautobot_device_onboarding/urls.py", line 18, in <module>
    from .views import (
  File "/usr/local/lib/python3.7/site-packages/nautobot_device_onboarding/views.py", line 21, in <module>
    from .forms import OnboardingTaskForm, OnboardingTaskFilterForm, OnboardingTaskFeedCSVForm
  File "/usr/local/lib/python3.7/site-packages/nautobot_device_onboarding/forms.py", line 24, in <module>
    from .worker import enqueue_onboarding_task
  File "/usr/local/lib/python3.7/site-packages/nautobot_device_onboarding/worker.py", line 28, in <module>
    from .onboard import OnboardingManager
  File "/usr/local/lib/python3.7/site-packages/nautobot_device_onboarding/onboard.py", line 17, in <module>
    from .netdev_keeper import NetdevKeeper
  File "/usr/local/lib/python3.7/site-packages/nautobot_device_onboarding/netdev_keeper.py", line 20, in <module>
    from napalm import get_network_driver
  File "/usr/local/lib/python3.7/site-packages/napalm/__init__.py", line 11, in <module>
    import napalm.iosxr
  File "/usr/local/lib/python3.7/site-packages/napalm/iosxr/__init__.py", line 22, in <module>
    from napalm.iosxr.iosxr import IOSXRDriver  # noqa
  File "/usr/local/lib/python3.7/site-packages/napalm/iosxr/iosxr.py", line 28, in <module>
    from napalm.pyIOSXR import IOSXR
  File "/usr/local/lib/python3.7/site-packages/napalm/pyIOSXR/__init__.py", line 26, in <module>
    from napalm.pyIOSXR.iosxr import IOSXR
  File "/usr/local/lib/python3.7/site-packages/napalm/pyIOSXR/iosxr.py", line 38, in <module>
    from netmiko.ssh_exception import NetMikoTimeoutException
ModuleNotFoundError: No module named 'netmiko.ssh_exception'
```

Workaround add `netmiko = ^3.0.0` to your local pyproject for Nautobot.

**Comment by mzbroch** [comment]
Thanks @nniehoff , I believe the issue raised because of:
- https://github.com/nautobot/nornir-nautobot/issues/55#issuecomment-1199007878 (nornir-nautobot installs nornir-netmiko 0.2)
- For some reason, poetry did not restrict netmiko to be lower than 4. I will review the reason of this behaviour

**Comment by mzbroch** [comment]
Fixed in #45
